packages/ui/components/input-date/src/LionInputDate.js

```diff
@@ -2,6 +2,32 @@ import { IsDate } from '@lion/ui/form-core.js';
 import { LionInput } from '@lion/ui/input.js';
 import { formatDate, LocalizeMixin, parseDate } from '@lion/ui/localize-no-side-effects.js';
 
+/**
+ *
+ * @param {Date | undefined} newV
+ * @param {Date | undefined} oldV
+ */
+function hasDateChanged(newV, oldV) {
+  return !(
+    (
+      Object.prototype.toString.call(newV) === '[object Date]' && // is Date Object
+      newV &&
+      newV.getDate() === oldV?.getDate?.() && // day
+      newV.getMonth() === oldV.getMonth?.() && // month
+      newV.getFullYear() === oldV.getFullYear?.()
+    ) // year
+  );
+}
+
+/**
+ * Change function that returns true if `value` is different from `oldValue`.
+ * This method is used as the default for a property's `hasChanged` function.
+ */
+export const notEqual = (value, old) => {
```

Comment:
TODO: in latest lit, we can import this function: https://github.com/lit/lit/blob/main/packages/reactive-element/src/reactive-element.ts#L376